### ISSUE TITLE
MT1012 - False Positive

### ISSUE THREAD

**Issue by nnpcYvIVl** [smartanalyzers/MultithreadingAnalyzer#5]
The following code throws MT1012 `Acquiring lock should always be wrapped in try block to ensure execution`. Is this a false positive?

```
using (AutoResetEvent autoresetevent = new AutoResetEvent(true))
{
    autoresetevent.WaitOne();
}
```

**Comment by cezarypiatek** [comment]
Thanks for reporting this issue. It looks like the current logic of `MT1012` and `MT1013` is too naive. For now, please suppress these two rules (or decrease the severity). I will let you know when I improve this diagnostic.

**Comment by cezarypiatek** [comment]
@nnpcYvIVl  I've fixed the problem with reporting MT1012 and MT1013 for signaling primitives. Please try the new version. I will keep this issue open for the reference because  MT1012 and MT1013  required additional adjustments.

**Comment by nnpcYvIVl** [comment]
@cezarypiatek Thanks, the new version works fine.
